scalahost/core/src/main/scala/scala/meta/Mirror.scala

```diff
@@ -1,16 +1,53 @@
 package scala.meta
 
-import scala.{Seq => _}
 import scala.collection.immutable.Seq
+import scala.meta.internal.scalahost.v1.offline
+import scala.meta.internal.scalahost.v1.online
 import scala.tools.nsc.Global
-import scala.meta.internal.scalahost.v1.online.{Mirror => OnlineMirror}
-import scala.meta.internal.scalahost.v1.offline.{Mirror => OfflineMirror}
-import scala.util.Try
 
 object Mirror {
-  def apply(global: Global): Mirror = new OnlineMirror(global)
-  def apply(classpath: String, sourcepath: String): Mirror =
-    new OfflineMirror(classpath, sourcepath)
+
+  /** Construct online Mirror from scalac Global instance.
+    *
+    * You may wish to use this constructor from within, for example:
+    * - a compiler plugin
+    * - the presentation compiler
+    */
+  def apply(global: Global): Mirror = new online.Mirror(global)
+
+  /** Construct offline Mirror from a persisted semantic database.
+    *
+    * @param classpath java.io.File.pathSeparator separated list of jar files
+    *                 or directories containing classfiles and `semanticdb` files.
+    *                 The `semanticdb` files are emitted by the scalahost-nsc
```

Comment:
`scalahost-nsc` is the compiler plugin that emits semanticdb. `scalahost` is the core library used by `scalahost-nsc` AND used by libraries like scalafix that construct the offline mirror.